lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -199,6 +213,53 @@ fetchDelistedPools tr uri manager = runExceptTLog $ do
                 , show s
                 ]
 
+    fromHttpException :: Monad m => HttpException -> m (Either String a)
+    fromHttpException = return . Left . ("HTTP exception: " <>) . show
+
+-- | Gets the health status from the SMASH server. Returns
+-- @Nothing@ if the server is unreachable.
+healthCheck
+    :: Maybe (Tracer IO StakePoolMetadataFetchLog)
+    -> URI
+    -> Manager
+    -> IO (Maybe HealthStatusSMASH)
+healthCheck mtr uri manager = runExceptTLog $ do
+    pl <- smashRequest mtr
+        (uri { uriPath = "/" <> healthCheckEP , uriQuery = "", uriFragment = "" })
+        manager
+    except . eitherDecodeStrict @HealthStatusSMASH $ pl
+  where
+    runExceptTLog
+        :: ExceptT String IO HealthStatusSMASH
+        -> IO (Maybe HealthStatusSMASH)
+    runExceptTLog action = runExceptT action >>= \case
+        Left msg ->
+            Nothing <$ trMaybe mtr (MsgFetchHealthCheckFailure msg)
+
+        Right health
+            | isHealthyStatus health -> do
+                trMaybe mtr (MsgFetchHealthCheckFailure
+                    ("Server reports unhealthy status: " <> T.unpack (status health)))
```

Comment:
It's better to do log message formatting in the ToText StakePoolMetadataFetchLog instance.
Then the program logic is not obscured by logging.